Std/Control/State.lean

```diff
@@ -0,0 +1,11 @@
+/-
+Copyright (c) 2023 James Gallicchio.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: James Gallicchio
+-/
+
+instance [Monad m] : MonadLift (StateM σ) (StateT σ m) where
+  monadLift s := pure ∘ s
+
+instance [MonadLift m n] : MonadLift (StateT σ m) (StateT σ n) where
+  monadLift s := fun state => liftM (s state)
```

Comment:
Does this have performance or error message implications?  I think `MonadLift` is intended from outer to inner while this would potentially enable a search process.